table/tables/index.go

```diff
@@ -301,8 +313,12 @@ func (c *index) Exist(sc *stmtctx.StatementContext, rm kv.RetrieverMutator, inde
 	return true, h, nil
 }
 
-func (c *index) FetchValues(r []types.Datum) ([]types.Datum, error) {
-	vals := make([]types.Datum, len(c.idxInfo.Columns))
+func (c *index) FetchValues(r []types.Datum, vals []types.Datum) ([]types.Datum, error) {
+	needLength := len(c.idxInfo.Columns)
+	if vals == nil || cap(vals) < needLength {
```

Comment:
```
if len(vals)  < needLength {
    vals = make([]types.Datum, needLength)
}
```

if 
cap(vals) = 100
len(vals) = 1
needLength = 10
line 321 will panic.

Reply:
```
package main

import "fmt"

func main() {
	vals := make([]int, 1, 100)
	val := vals[:10]
	fmt.Println(val)
}
```
Not panic...

Reply:
Wow, I did not expect Go allow this feature.

Reply:
Yes, golang support it.